neard/tests/run_nodes.rs

```diff
@@ -17,22 +19,24 @@ fn run_nodes(
 ) {
     let mut rng = thread_rng();
     let genesis_height = rng.gen_range(0, 10000);
-    run_actix_until_stop(async move {
-        let dirs = (0..num_nodes)
-            .map(|i| {
-                tempfile::Builder::new()
-                    .prefix(&format!("run_nodes_{}_{}_{}", num_nodes, num_validators, i))
-                    .tempdir()
-                    .unwrap()
-            })
-            .collect::<Vec<_>>();
-        let (_, _, clients) =
-            start_nodes(num_shards, &dirs, num_validators, 0, epoch_length, genesis_height);
+
+    let cluster = NodeCluster::new()
+        .mkdirs_with(num_nodes as usize, |index| {
+            format!("run_nodes_{}_{}_{}", num_nodes, num_validators, index)
+        })
```

Comment:
Let's just merge `mkdirs_with` into `new`. It does not make any sense to have a cluster without nodes at all.

```suggestion
    let cluster = NodeCluster::new(num_nodes as usize, |index| {
            format!("run_nodes_{}_{}_{}", num_nodes, num_validators, index)
        })
```

Reply:
😅, you're totally right. I'm on it.